docker/Dockerfile

```diff
@@ -1,71 +1,106 @@
+FROM cartesi/rollups-hardhat:1.0.0 as rollups-hardhat
 
-
-# download standard machine drives
+# stage to download dependencies
 FROM busybox as machine-core
 
 WORKDIR /opt/cartesi/share
 
-# download dependencies
 COPY dependencies .
 COPY shasumfile .
 RUN while read DEP; do wget -O $DEP; done < dependencies
 RUN sha1sum -c shasumfile
 
 
-# Deploy contracts locally
-FROM rollups-hardhat AS local-deployments-stage
+FROM debian:bookworm-20230814 as genext2fs
 
-RUN sh -c "npx hardhat deploy --write true --network hardhat"
-ENTRYPOINT [ "/bin/sh" ]
+WORKDIR /usr/local/src
+ARG GENEXT2FS_VERSION=1.5.2
 
-# Download known testnet/mainet deployments
-FROM busybox as deployments-stage
 
-WORKDIR /opt/cartesi/share
-COPY --from=machine-core opt/cartesi/share/rollups.tgz ./
-RUN tar -xzvf ./rollups.tgz
+RUN <<EOF
+apt-get update
+apt-get install -y --no-install-recommends autoconf automake build-essential ca-certificates curl libarchive-dev libtool
+rm -rf /var/lib/apt/lists/*
+EOF
+
 
-# stage to build the dapp ext2 filesystem
+RUN <<EOF
+curl -sL https://github.com/cartesi/genext2fs/archive/refs/tags/v${GENEXT2FS_VERSION}.tar.gz | tar --strip-components=1 -zxvf -
+./autogen.sh
+./configure --enable-libarchive
+make
+make install
+EOF
+
+# stage to wrap the dapp installing resources needed to run inside the Cartesi Machine
+FROM dapp as wrapped-stage
+
+WORKDIR /
+
+# normalize date for subsequent operations
+ENV SOURCE_DATE_EPOCH=1640995200
+
+# install busybox
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    busybox-static=1:1.30.1-7ubuntu3 \
+    && rm -rf /var/lib/apt/lists/* \
+    && find /var/log \( -name '*.log' -o -name '*.log.*' \) -exec truncate -s 0 {} \;
+
+# stage to build the final ext2 filesystem for the dapp
 FROM toolchain as fs-stage
 
-# install jq
+WORKDIR /var/tmp/dapp_fs/
+
+# copy wrapped dapp filesystem
+COPY --from=wrapped / ./
+
+# add emulator tools
+COPY --from=machine-core /opt/cartesi/share/machine-emulator-tools.tar.gz /var/tmp/dapp_fs/
+RUN tar -xvf machine-emulator-tools.tar.gz ./
+RUN rm machine-emulator-tools.tar.gz
+
+
+#add new genext2fs
+COPY --from=genext2fs /usr/local/bin/genext2fs /usr/bin/
+
+# enforce locale
 RUN <<EOF
 apt-get update
-DEBIAN_FRONTEND="noninteractive" apt-get install -y jq
-rm -rf /var/lib/apt/lists/*
+apt-get install -y --no-install-recommends locales
+sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen
+locale-gen
+update-locale LANG=en_US.UTF-8
 EOF
 
-WORKDIR /opt/cartesi/dapp
-COPY --from=dapp /opt/cartesi/dapp .
-COPY --from=deployments /opt/cartesi/share/package/deployments ./deployments
-COPY --from=local-deployments /app/rollups/deployments/hardhat/* ./deployments/localhost/
-
-COPY build-dapp-fs.sh /usr/local/bin
-COPY default.json /opt/cartesi/
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
 
-RUN build-dapp-fs.sh /opt/cartesi/default.json dapp.json dapp.ext2
```

Comment:
This file `default.json` still exists, but looks like outdated, it says ram length is 128Mb, but it is now 64Mb, I am not sure if you should delete it or update it.

Reply:
The file was deleted.